src/World.cpp

```diff
@@ -3309,7 +3360,15 @@ cPlayer * cWorld::FindClosestPlayer(Vector3d a_Pos, float a_SightLimit, bool a_C
 		ClosestDistance = Distance;
 		ClosestPlayer = *itr;
 	}
-	return ClosestPlayer;
+
+	if (ClosestPlayer)
+	{
+		return a_Callback(*ClosestPlayer);
```

Comment:
Should this return `!a_Callback(ClosestPlayer)` instead?  There are existing `DoWith` functions that use both conventions:

https://github.com/cuberite/cuberite/blob/c94d7184ebaf7e8540f717c70c1e03ae62e5a7bd/src/Chunk.cpp#L2279
https://github.com/cuberite/cuberite/blob/c94d7184ebaf7e8540f717c70c1e03ae62e5a7bd/src/World.cpp#L2807
... and a third for good measure...
https://github.com/cuberite/cuberite/blob/c94d7184ebaf7e8540f717c70c1e03ae62e5a7bd/src/World.cpp#L2753-L2754

Reply:
In my understanding,
callback in *ForEach..* method return false means continue, return true to abort.
But in *DoWith..* method, there is no such rule, so what the return value means depend on each method. 
some *DoWith..* don't care about callback return value, return true means callback had being called.

In method *DoWithNearestPlayer*, method return what callback returns, I think it is make sense.

Reply:
I have no doubt that in isolation all of these were justified but when taken together it makes the interface to `cWorld` inconsistent.  As far as I can tell, there are a lot more `DoWith` functions that try to behave like a single element `ForEach` function. So, for consistency, it might be better to go that way. 

I think it's at least worth discussing.

Reply:
Hi @peterbell10 

I agree we should refactor `ForEach` and `DoWith` method in `cWorld` (maybe in another pull request), there are lots of interface return style and comment inconsistent.

After read source code , here is my proposal:
1. for every `ForEach` method , 
    * callback return **true** to abort current for each loop, return **false** to continue loop.
    * 'ForEach' method return **true** if all element is processed, return **false** if abort by callback.

2. for every `DoWith` method,
    * `DoWith` return **false** if callback is not executed for some reason.
    * if callback executed, `DoWith` return whatever callback returns, so that it is easy to get callback return value from `DoWith` return value.

Reply:
The interface can't really be changed because the functions are exported to the Lua API. Which is why new interfaces should try to be consistent with the old.

Reply:
What @changyongGuo describes is the general idea I had when designing those functions from the start, if they behave different anywhere, it should be considered a bug and fixed, even if it means breaking plugins in the process (I don't think there are that many plugins that use those APIs, yet, so it should be fine). But such a change is for another PR, this PR should add the function so that it adheres to this API.

I'll check all the `DoWith` functions and make a list of those that need changing, as a separate issue.